### ISSUE TITLE
Unsound annotations in "Class.java"

### ISSUE THREAD

**Issue by jwaataja** [t-rasmud/checker-framework#151]
In `Class.java`, there are many methods which return `@Det` which could be non-deterministic given a non-deterministic receiver. Take, for example, these method annotations:
```java
@Det boolean isInterface(@PolyDet Class<T> this);
@Det String getSimpleName(@PolyDet Class<T> this);
@Det Field @OrderNonDet [] getFields(@PolyDet Class<T> this);
```

If there are nondeterministic `Class<?>` objects, these methods could return non-deterministic results, so these return types should be `@PolyDet`.

**Comment by mernst** [comment]
This is correct.  I suspect the annotations were incorrectly copied from other type systems where the qualifier on a `Class` is irrelevant.

**Comment by t-rasmud** [comment]
Fixed in 2b329f2cd6d1130ce47bfe4a2740e0a999ee1f88
